### ISSUE TITLE
File size limits

### ISSUE THREAD

**Issue by SR5WKG4MWKQFWGS54AUE** [kgretzky/pwndrop#3]
I tried to upload files but most "bigger" ones failed. A couple hundred byte file size is okay but above ~30MB it starts the upload and the entry just disappears. Firefox 75/MacOS. I didn't do rigorous testing, maybe it's a problem on my end, was curious if this is a known limitation or case-by-case.

**Comment by kgretzky** [comment]
How fast was the download? Maybe you've experienced a timeout? Do you know approximately after how much time it fails from the start of the upload?

I've successfully tested with even 2GB uploads, so large files definitely are supported.

**Comment by principalarchivist** [comment]
Generally a delightful package, but I'm having the same problem as the prior poster: small files upload without an issue, large (in my case, 2.8GB, more or less) files bomb out before completion.  

I'm hosting pwndrop in a docker container on a Synology Diskstation on my LAN, and trying to copy a 2.8GB file hosted on that same Diskstation into pwndrop.  I have also tried downloading the file to my WIndows machine, and uploading the file into pwndrop from there.

The uploads do not always bomb out at the same point, but typically about a quarter to halfway through.  I've tried with Chrome, FIrefox (seemed to get farther than Chrome), and Edge and IE (neither of which worked at all).

Everything is connected by 1GB ethernet, and the network is otherwise solid; I don't have similar issues with other software.

A few things that are definitely NOT the issue:  (1) physical memory on the Windows machine; it's not filling up as the task progresses; (2) diskspace on the Diskstation...it has terabytes free... 

Happy to run any tests you might be able to think of... would really love to get this working!

**Comment by jdgvt81** [comment]
I'm running into the same problem, trying to upload a 16 MB and 24 MB file - but I'm running on running on a slower connection - 12 Mbps down, 3 Mbps up.  Progress bar gets 10-15% and the upload simply disappears.  Using Firefox 77.0.1.  Running pwndrop 1.0.0 hosted on a SSDNodes VPS using the Docker image behind a Traefik reverse proxy.  Smaller files uploaded ok.  Docker logs don't give any indication of an issue.  Is there any other information I can provide to help diagnose?

**Comment by N0ne1eft** [comment]
Same issue here, when trying to upload large files (about bigger than 100MB) the progress bar suddenly disappear at 10-15% progress. Safari console suggests the network connection was lost to **http://xx.xx.xx.xx/api/v1/files** when the upload disappears. Environment: pre-built binary from release running on docker container ubuntu:latest. Smaller files works really well without any problem.

**Comment by kgretzky** [comment]
Thanks for reporting guys. Can you use a stopwatch and measure if the upload gets cut off after exactly 15 seconds since the upload started?

This is how the `http.Server` is set up for pwndrop and I'm pretty sure it is breaking all connections whenever they reach 15 seconds timeouts:
```
	s.srv = &http.Server{
		Handler:      http.Handler(s),
		Addr:         hostname,
		WriteTimeout: 15 * time.Second,
		ReadTimeout:  15 * time.Second,
		TLSConfig:    tls_cfg,
	}
```

There is a good post on the issues involved here: [https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/](url)

I'm trying to figure out how to set it up properly. Usually only authenticated users will be uploading files, so I may increase the timeouts without risking susceptibility to denial of service attacks.

**Comment by N0ne1eft** [comment]
> Thanks for reporting guys. Can you use a stopwatch and measure if the upload gets cut off after exactly 15 seconds since the upload started?
> 
> This is how the `http.Server` is set up for pwndrop and I'm pretty sure it is breaking all connections whenever they reach 15 seconds timeouts:
> 
> ```
> 	s.srv = &http.Server{
> 		Handler:      http.Handler(s),
> 		Addr:         hostname,
> 		WriteTimeout: 15 * time.Second,
> 		ReadTimeout:  15 * time.Second,
> 		TLSConfig:    tls_cfg,
> 	}
> ```
> 
> There is a good post on the issues involved here: [https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/](url)
> 
> I'm trying to figure out how to set it up properly. Usually only authenticated users will be uploading files, so I may increase the timeouts without risking susceptibility to denial of service attacks.

Yes I did few tests using large files and it did cut off exactly 15 seconds after upload started. Therefore the upload size is somehow limited by upload speed*15 seconds.

**Comment by kgretzky** [comment]
Just pushed a commit increasing the timeouts: https://github.com/kgretzky/pwndrop/commit/5f9566c55bc9e802327178909eb7f084c0d6f2dc

Can you test the changes by compiling from source or do you need me to post a binary release?

**Comment by principalarchivist** [comment]
I'm afraid I need a binary release.  :(

**Comment by N0ne1eft** [comment]
> Just pushed a commit increasing the timeouts: [5f9566c](https://github.com/kgretzky/pwndrop/commit/5f9566c55bc9e802327178909eb7f084c0d6f2dc)
> 
> Can you test the changes by compiling from source or do you need me to post a binary release?

Tested, the issue has been resolved and large file can be uploaded without any error. Thanks for the quick update.

**Comment by orangethakkali** [comment]
Does the latest docker have this fix.

**Comment by jankaifer** [comment]
I am trying to upload a 700MB file, and after 2 mins I get `413 Request Entity Too Large`.
Probably different cause but same issue.

Sorry, that error was thrown by Nginx. solved by adding:
```
client_max_body_size 2G;
```

It works now with a 700M video.

**Comment by damajor** [comment]
When uploading large files the tmp directory is not cleaned on my system. multipart-1234567 files are still there.
Do you experience the same issue ?
